### PR TITLE
[graphql] Add coinConnection to Query

### PR DESCRIPTION
**Pull request by stefan-mysten**
## Description 

Adds `coinConnection` to top-level `Query`

## Test Plan 

Need to figure out how to write tests for this. Until then, here's a screenshot.

image

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-graphql-rpc/schema/current_progress_schema.graphql [hunk 1]
```diff
@@ -810,7 +810,7 @@ type Owner implements ObjectOwner {
 	"""
 	coinConnection(first: Int, after: String, last: Int, before: String, type: String): CoinConnection
 	"""
-	The stake objects for the given address
+	The `0x3::staking_pool::StakedSui` objects owned by the given object.
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/schema/current_progress_schema.graphql:813]: Probably should have done a separate PR, but I saw that this is one place where the comment is different than the other two existing `stakedSuiConnection` comments, so decided to update it.

Comment (amnn) [reply crates/sui-graphql-rpc/schema/current_progress_schema.graphql:813]: I'll let it slide ;) thanks for the clean-up!

File: crates/sui-graphql-rpc/schema/current_progress_schema.graphql [hunk 2]
```diff
@@ -907,6 +907,12 @@ type Query {
 	epoch(id: Int): Epoch
 	checkpoint(id: CheckpointId): Checkpoint
 	transactionBlock(digest: String!): TransactionBlock
+	"""
+	The coin objects that exist in the network.
+	The type field is a string of the inner type of the coin
+	by which to filter (e.g., 0x2::sui::SUI).
```
Comment (amnn) [inline crates/sui-graphql-rpc/schema/current_progress_schema.graphql:913]: Does `0x2::sui::SUI` actually work as a filter, or do you need to canonicalize it before passing it in (for now)?

I don't mind if you have to canonicalize it, initially, but just want to make sure that the doc comment is not suggesting an example that won't work.

Comment (amnn) [reply crates/sui-graphql-rpc/schema/current_progress_schema.graphql:913]: It also seems like `type` is an optional parameter.  Could you augment the docs to say what happens when you omit it?

Comment (amnn) [reply crates/sui-graphql-rpc/schema/current_progress_schema.graphql:913]: Final thought (this one we can follow-up on): Typically, with the other connections, (`checkpointConnection` is the odd one out here), we offer a way, with the `Query` version of the connection, to write a generic query -- one that supports any kind of query you could have constructed through some number of nested queries. For `coin` this means introducing a `CoinFilter` that allows folks to optionally supply the address as well as the type.  For `checkpointConnection` it involves adding an optional filter by epoch.

**Review by amnn — APPROVED** [review]
Looks great, thank you!  Some nits on docs, and a follow-up for us to record as well.

File: crates/sui-graphql-rpc/schema/current_progress_schema.graphql (hunk 2 above)
Comment (stefan-mysten) [reply crates/sui-graphql-rpc/schema/current_progress_schema.graphql:913]: I'll do a follow up PR on the `CoinFilter`, that's a good idea.

Comment (stefan-mysten) [reply crates/sui-graphql-rpc/schema/current_progress_schema.graphql:913]: > Does `0x2::sui::SUI` actually work as a filter, or do you need to canonicalize it before passing it in (for now)?
> 
> I don't mind if you have to canonicalize it, initially, but just want to make sure that the doc comment is not suggesting an example that won't work.

It does not work as in the comment. I will update and then try to update it everywhere else. Then we'll have to update it again to mention that we support both shorthand & canonical.
